server/src/main/java/org/elasticsearch/common/time/EpochMillisDateFormatter.java

```diff
@@ -70,4 +89,31 @@ public String pattern() {
     public DateFormatter parseDefaulting(Map<TemporalField, Long> fields) {
         return this;
     }
+
+    @Override
+    public Locale getLocale() {
+        return locale;
+    }
+
+    @Override
+    public ZoneId getZone() {
+        return zoneId;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(locale);
```

Comment:
Depending on whether zoneId should be kept (if it is always UTC, maybe not?), maybe include it in the hashCode as well? Although I think it is not strictly necessary, since it is only necessary that equal objects must have equal hash codes, not the other way round.